blivet/fstab.py

```diff
@@ -0,0 +1,792 @@
+
+    def _copy_fs_entry(self, entry):
+        """ Create copy of libmount.Fs()
+        """
+        # Nope, it has to be done like this. Oh well...
+        new_entry = Fs()
+        new_entry.source = entry.source
+        new_entry.target = entry.target
+        new_entry.fstype = entry.fstype
+        new_entry.append_options(entry.options)
+        new_entry.freq = entry.freq
+        new_entry.passno = entry.passno
+        if entry.comment is not None:
+            new_entry.comment = entry.comment
+        return new_entry
+
+    def _parser_errcb(self, tb, fname, line):  # pylint: disable=unused-argument
+        """ Libmount interface error reporting function
+        """
+        log.error("Fstab parse error '%s:%s'", fname, line)
+        return 1
+
+    def entry_from_device(self, device):
+        """ Generate FSTabEntry object based on given blivet Device
+
+            :keyword device: device to process
+            :type device: :class: `blivet.devices.StorageDevice`
+            :returns: fstab entry object based on device
+            :rtype: :class: `FSTabEntry`
+        """
+
+        entry = FSTabEntry()
+
+        if device.format.uuid:
+            entry.spec = "UUID=%s" % device.format.uuid
+        else:
+            entry.spec = getattr(device, "fstab_spec", None)
+
+        entry.file = None
+        if device.format.mountable:
+            entry.file = device.format.mountpoint
+        if device.format.type == "swap":
+            entry.file = "swap"
+
+        entry.vfstype = device.format.type
+
+        return entry
+
+    def read(self, src_file=""):
+        """ Read the fstab file from path stored in self.src_file. Setting src_file parameter overrides
+            it with new value.
+
+            :keyword src_file: When set, reads fstab from path specified in it
+            :type src_file: str
+        """
+
+        # Reset table
+        self._table = Table()
+        self._table.enable_comments(True)
+
+        # resolve which file to read
+        if src_file == "":
+            if self.src_file is None:
+                # No parameter given, no internal value
+                return
+        elif not src_file:
+            return
+        else:
+            self.src_file = src_file
+
+        self._table.errcb = self._parser_errcb
+        try:
+            self._table.parse_fstab(self.src_file)
+        except Exception as e:  # pylint: disable=broad-except
+            # libmount throws general Exception. Okay...
+            if str(e) == "No such file or directory":
+                log.warning("Fstab file '%s' does not exist", self.src_file)
+            else:
+                raise
+
+    def find_device(self, blivet, spec=None, mntops=None, blkid_tab=None, crypt_tab=None, *, entry=None):
+        """ Find a blivet device, based on spec or entry. Mount options can be used to refine the search.
+            If both entry and spec/mntops are given, spec/mntops are prioritized over entry values.
```

Comment:
I'd prefer having two functions here instead of this. For example something like `find_device_by_entry` and `find_device_by_spec`.

Reply:
Combining these functions together allows finding the device based on multiple criteria if needed.